src/interface/window.c

```diff
@@ -1445,6 +1445,49 @@ void window_rotate_camera(rct_window *w, int direction)
 	reset_all_sprite_quadrant_placements();
 }
 
+void window_viewport_get_map_coords_by_cursor(rct_window *w, sint16 *map_x, sint16 *map_y, sint16 *offset_x, sint16 *offset_y)
+{
+	// Get mouse position to offset against.
+	int mouse_x, mouse_y;
+	platform_get_cursor_position_scaled(&mouse_x, &mouse_y);
+
+	// Compute map coordinate by mouse position.
+	get_map_coordinates_from_pos(mouse_x, mouse_y, VIEWPORT_INTERACTION_MASK_NONE, map_x, map_y, NULL, NULL, NULL);
+
+	// Get viewport coordinates centring around the tile.
+	int base_height = map_element_height(*map_x, *map_y);
+	int dest_x, dest_y;
+	center_2d_coordinates(*map_x, *map_y, base_height, &dest_x, &dest_y, w->viewport);
+
+	// Rebase mouse position onto centre of window, and compensate for zoom level.
+	int rebased_x = ((w->width >> 1) - mouse_x) << w->viewport->zoom,
```

Comment:
@janisozaur Right, I see what you mean. You'd rather I use simple multiplication and division here, then?